### ISSUE TITLE
Add grey option when disabling checkboxs in dark mode

### ISSUE THREAD

**Issue by BennyBaruba** [Esri/calcite-design-system#6823]
### Description

In the Mission Manager app in ArcGIS Mission, we are finding that a disabled checkbox is hard to distinguish from the enabled checkbox in dark mode.  See image.

![image](https://user-images.githubusercontent.com/5122225/233635990-4d8e2c9f-1e83-47d6-8636-dad161ee5af3.png)

We would like the option to use a grey checkbox to show that a checkbox is disabled.  See image.

![image](https://user-images.githubusercontent.com/5122225/233636278-fd387a34-66c6-454c-8d87-d511fab303b8.png)

### Acceptance Criteria

- [ ] A grey color can be used to show a disabled checkbox

### Which Component

Checkbox

### Priority impact

p4 - not time sensitive

### Esri team

ArcGIS Mission

**monday.com sync:** #9363632005

**Comment by macandcheese** [comment]
You can theme the disabled and checked state of the Combobox to achieve this: 

https://codepen.io/mac_and_cheese/pen/rNqjNMY?editors=1000

**Comment by macandcheese** [comment]
cc @ashetland @SkyeSeitz - is this a change we want to make to the component?

**Comment by ashetland** [comment]
Since this can be achieved through theming, I don't think it's necessary. I would also note that, when comparing the contrast of disabled to enabled, there appears to be even less of a contrast difference between the two when using grey in the screenshots above. That said, the un-themed version could be better too. Might be worth taking a holistic look at this.

**Comment by SkyeSeitz** [comment]
I agree, I would be more inclined to solve the opacity approach and ensure it is effective in dark mode.

**Comment by macandcheese** [comment]
Let's consider closing this as unplanned - this styling remains possible and is pretty simple to achieve with css properties cc @Esri/calcite-pes @Esri/calcite-designers

**Comment by geospatialem** [comment]
> Let's consider closing this as unplanned - this styling remains possible and is pretty simple to achieve with css properties cc [@Esri/calcite-pes](https://github.com/orgs/Esri/teams/calcite-pes) [@Esri/calcite-designers](https://github.com/orgs/Esri/teams/calcite-designers)


Closing the above out as unplanned with support through styling, such as https://codepen.io/geospatialem/pen/myEOGzw:

```css
  .disabled-checkbox {
    --calcite-checkbox-border-color: lightgray;
    --calcite-checkbox-icon-color: lightgray;
    --calcite-color-brand: darkgray;
  }
```
